dataset/src/test/scala/frameless/CreateTests.scala

```diff
@@ -28,6 +28,66 @@ class CreateTests extends TypedDatasetSuite with Matchers {
       Vector[(Food, Country)]] _))
   }
 
+  test("array fields") {
+
+    def prop[T](implicit arb: Arbitrary[Array[T]], encoder: TypedEncoder[X1[Array[T]]]) = forAll {
+      data: Array[T] =>
+        val Seq(X1(arr)) = TypedDataset.create(Seq(X1(data))).collect().run()
+        Prop(arr.sameElements(data))
+    }
+
+    check(prop[Boolean])
+    check(prop[Byte])
+    check(prop[Short])
+    check(prop[Int])
+    check(prop[Long])
+    check(prop[Float])
+    check(prop[Double])
+    check(prop[X1[String]])
+    check(prop[String])
+
+  }
+
+  test("vector fields") {
+
+    def prop[T](implicit arb: Arbitrary[Vector[T]], encoder: TypedEncoder[X1[Vector[T]]]) = forAll {
+      data: Vector[T] =>
+        val Seq(X1(vec)) = TypedDataset.create(Seq(X1(data))).collect().run()
+        Prop(vec == data)
+    }
+
+    check(prop[Boolean])
+    check(prop[Byte])
+    check(prop[Short])
+    check(prop[Int])
+    check(prop[Long])
+    check(prop[Float])
+    check(prop[Double])
+    check(prop[X1[String]])
+    check(prop[String])
+
+  }
+
+  test("map fields (scala.Predef.Map / scala.collection.immutable.Map)") {
+
+    def prop[A, B](implicit arb: Arbitrary[Map[A, B]], encoder: TypedEncoder[X1[Map[A, B]]]) = forAll {
+      data: Map[A, B] =>
+        val Seq(X1(map)) = TypedDataset.create(Seq(X1(data))).collect().run()
+        Prop(map == data)
+    }
+
+    check(prop[String, Boolean])
+    check(prop[String, Byte])
+    check(prop[String, Short])
+    check(prop[String, Int])
+    check(prop[String, Long])
+    check(prop[String, Float])
+    check(prop[String, Double])
+    check(prop[String, X1[String]])
+    check(prop[String, String])
```

Comment:
What about tests where:
1. key isn't `String`
2. key is `Option[_]`
3. key is `X1[_]`
4. value is `Option[_]`